### DISCUSSION TITLE
[@types/mdx] `FunctionComponent<T>` is incompatible with `FC<T>` in `@types/react` after #65135

### DISCUSSION THREAD

**Discussion by loxygenK** [DefinitelyTyped/DefinitelyTyped#65745] [Issues with a @types package] [answered]
By the modification in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65135 that `@types/react` newly accepts `ReactNode` as the return from `FC<T>`. However, type `FunctionComponent<T>` defined privately in `@types/mdx` uses `JSX.Element | null`,  so types `FC<T>` and `FunctionComponent<T>` became incompatible. Due to this, passing the component to `<MDXProvider>` as props now raise a type error if the component is declared as type `FC<T>`.

```tsx
import { FC, ReactNode } from 'react';
import { MDXProvider } from '@mdx-js/react';

// removing `FC<{ children: ReactNode }>` resolves the issue
export const Heading1: FC<{ children: ReactNode }> = ({ children }) => {
  return <h1>H1: {children}</h1>;
};

export const Provider: FC<{ children: ReactNode }> = ({ children }) => {
  return <MDXProvider components={{ h1: Heading1 }}>{children}</MDXProvider>;
};
```

```bash
❯ pnpm tsc
index.tsx:10:37 - error TS2322: Type 'FC<{ children: ReactNode; }>' is not assignable to type 'Component<DetailedHTMLProps<HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>>'.
  Type 'FunctionComponent<{ children: ReactNode; }>' is not assignable to type 'FunctionComponent<DetailedHTMLProps<HTMLAttributes<HTMLHeadingElement>, HTMLHeadingElement>>'.
    Type 'ReactNode' is not assignable to type 'Element'.
      Type 'string' is not assignable to type 'ReactElement<any, any>'.

10   return <MDXProvider components={{ h1: Heading1 }}>{children}</MDXProvider>;
                                       ~~
```

Reproduction is here: https://stackblitz.com/edit/types-mdx-type-issue-repro?file=index.tsx

[a comment by an automated account is not shown]

**Comment by wooorm** [comment, marked as answer]
Did you see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65703? That might solve it.

**Comment by loxygenK** [comment]
Thanks for the reply! I didn't see this...
The PR solved the error! I ran the type check on this PR's branch and no error was raised.

Should I close this discussion or keep it open until the PR is merged?

**Reply by wooorm** [reply]
I dunno what TS folks prefer, but maybe a review there might be useful!

**Comment by remcohaszing** [comment]
Fixed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65703
